py/torch_tensorrt/dynamo/_compiler.py

```diff
@@ -319,3 +325,122 @@ def compile_module(
+        else:
+            output_dtypes.append(output.dtype)
+
+    interpreter = TRTInterpreter(
+        module,
+        inputs,
+        logger_level=(trt.Logger.VERBOSE if settings.debug else trt.Logger.WARNING),
+        output_dtypes=output_dtypes,
+        compilation_settings=settings,
+    )
+    interpreter_result = interpreter.run(
+        workspace_size=settings.workspace_size,
+        precision=settings.precision,
+        profiling_verbosity=(
+            trt.ProfilingVerbosity.VERBOSE
+            if settings.debug
+            else trt.ProfilingVerbosity.LAYER_NAMES_ONLY
+        ),
+        max_aux_streams=settings.max_aux_streams,
+        version_compatible=settings.version_compatible,
+        optimization_level=settings.optimization_level,
+    )
+    return interpreter_result
+
+
+def convert_method_to_trt_engine(
+    module: torch.fx.GraphModule,
+    method_name: str = "forward",
+    inputs: Optional[Sequence[Input | torch.Tensor]] = None,
+    device: Device = Device._current_device(),
+    disable_tf32: bool = False,
+    sparse_weights: bool = False,
+    enabled_precisions: Optional[Set[torch.dtype | _enums.dtype]] = None,
+    refit: bool = False,
+    debug: bool = False,
+    capability: _enums.EngineCapability = _enums.EngineCapability.default,
+    num_avg_timing_iters: int = 1,
+    workspace_size: int = 0,
+    dla_sram_size: int = 1048576,
+    dla_local_dram_size: int = 1073741824,
+    dla_global_dram_size: int = 536870912,
+    truncate_long_and_double: int = False,
+    calibrator: object = None,
+    allow_shape_tensors: bool = False,
+) -> bytes:
+    if debug:
+        set_log_level(logger.parent, logging.DEBUG)
+
+    input_list = list(inputs) if inputs is not None else []
+    # Prepare torch_trt inputs
+    input_list = prepare_inputs(input_list)
+    device = to_torch_tensorrt_device(device)
+
+    enabled_precisions = (
+        enabled_precisions if enabled_precisions is not None else {torch.float}
+    )
+
+    if (
+        torch.float16 in enabled_precisions
+        or torch_tensorrt.dtype.half in enabled_precisions
+    ):
+        precision = torch.float16
+    elif (
+        torch.float32 in enabled_precisions
+        or torch_tensorrt.dtype.float in enabled_precisions
+    ):
+        precision = torch.float32
+    elif len(enabled_precisions) == 0:
+        logger.info(f"No precision specified, defaulting to {PRECISION}")
+        precision = PRECISION
+    else:
+        raise ValueError(
+            f"Precision {enabled_precisions} not supported in the Dynamo Path"
+        )
+
+    compilation_options = {
+        "precision": precision,
+        "debug": debug,
+        "device": device,
+        "workspace_size": workspace_size,
+        "truncate_long_and_double": truncate_long_and_double,
+        "max_aux_streams": MAX_AUX_STREAMS,
+        "version_compatible": VERSION_COMPATIBLE,
+        "optimization_level": OPTIMIZATION_LEVEL,
+    }
```

Comment:
This should use the same set of compilation options as here, perhaps omitting `use_python_runtime` and `use_fast_partitioner` which would not be necessary in the full-graph conversion case:
https://github.com/pytorch/TensorRT/blob/867dc7bf23b34f8d3809a1d64841b0013a9a0054/py/torch_tensorrt/dynamo/_compiler.py#L177-L195

Reply:
Thanks for the suggestion! Yes I did like this at first, but many arguments like `min_block_size, torch_executed_ops, pass_through_build_failures, etc.` don't show up in the function. Naren told me to use the function declaration from torchscript. So do you think we need to add more args in the declaration and remove those useless args?

Reply:
I see - the main concern is that `max_aux_streams`, `optimization_level`, and `version_compatible` options are all attributes of the TRT Engine itself (set in the `TRTInterpreter`), so the user should be able to customize these instead of just using the defaults.

Reply:
I agree with you! Maybe we can expose some/all args in the function declaration for users allowing them to specify values?

Reply:
I think just those three arguments suffice to expose in the function header